teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -0,0 +1,341 @@
+
+3. A dedicated HTTP Service
+
+  - Potential Auth and HA problems
+  - Could run as part of the controller
+  - Could be a separate HTTP server(s) which write to TaskRuns (or even config maps); task pods connect to this server to submit results, this server records the results (means result size would be limited to what can be stored in the CRD but there probably needs to be an upper bound on result size anyway)
+
+4. Self-update / mutate the TaskRun via admission controller (with the various controls i.e. first write, subsequent read only)
+
+  - Potential issue with self updating its own Status
+
+5. Separate Database
+
+  - Introducing an additional database requirement to Tekton to support the storage of information outside of etcd.
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+There is the current alternative of storing result parameters as data in a workspace, however Workspaces
+- require there has to be a storage mechanism in the cluster that can be shared between Tasks. That can be complex, or have performance issues in itself if using an As A Service that orders the storage at spin-up time. Or forces Tasks to all run on the same node. etc. Storage is a big complex adoption hurdle.
+- changes the way end users refer to the result parameter or pass between containers
+- requires some tasks to be altered to retrieve data from the file system in a certain location. This makes it difficult to use a library of Tekton Tasks or an abstraction that doesn't provide access to where a parameter comes from.
+
+## Infrastructure Needed (optional)
+
+<!--
+Use this section if you need things from the project/SIG.  Examples include a
+new subproject, repos requested, github details.  Listing these here allows a
+SIG to get the process for these resources started right away.
+-->
+
+## Upgrade & Migration Strategy (optional)
```

Comment:
Results today are stored are available via `TaskRun` and `PipelineRun` status, which is part of the Tekton API.
We should try and keep that interface even for storage of larger results, perhaps replacing the result text itself with a link to where the result can be obtained from.

Tekton can send `CloudEvents` and those events today include results. This mechanism is used in our dogfoding to send data to pipelines that are triggered asynchronously through events. I think this is a good use case, and we should allow for it to be possible regardless of the size and type of results that we support, with the caveat that larger results would not be directly in the status, only a link to them.